### PR TITLE
Add typings for anchor-js

### PR DESCRIPTION
**Pull request by xt0rted**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not provide its own types, and you can not add them.
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

Since I last added a typing the process has changed, so I followed the instructions using `dts-gen` but had a couple issues with it. The first was the `tsconfig.json` didn't include `"strictFunctionTypes": true,` so linting failed right away. The second was the generated `tslint.json` didn't include any rules so linting was also failing until I coppied the rules from another folder. I'm not sure if these things are safe defaults/required or not, but if they are it'd be nice if they were included when using `dts-gen` since they're extra manual steps.

### CODE REVIEW

File: types/anchor-js/tslint.json [hunk 1]
```diff
@@ -0,0 +1,79 @@
+{
+    "extends": "dtslint/dt.json",
```
Comment (plantain-00) [inline types/anchor-js/tslint.json]: For new packages, this file should only `"extends": "dtslint/dt.json"`

**Review by plantain-00 — CHANGES_REQUESTED** [review]

File: types/anchor-js/tslint.json (hunk 1 above)
Comment (xt0rted) [reply types/anchor-js/tslint.json]: Without the extra rules the linter kept complaining about the `declare module` at the bottom but its recommendations weren't working for me. How should this be written?

```typescript
declare module 'anchor-js' {
    export = AnchorJS;
}
```

Comment (plantain-00) [reply types/anchor-js/tslint.json]: try just `export = AnchorJS;` ?

Comment (xt0rted) [reply types/anchor-js/tslint.json]: Doing that gives errors in the test file.
```
Error: c:/dev/GitHub/DefinitelyTyped/types/anchor-js/anchor-js-tests.ts:1:1
ERROR: 1:1    expect  TypeScript@next compile error:
Cannot find name 'anchors'.
ERROR: 2:1    expect  TypeScript@next compile error:
Cannot find name 'anchors'.
ERROR: 3:1    expect  TypeScript@next compile error:
Cannot find name 'anchors'.
ERROR: 5:1    expect  TypeScript@next compile error:
Cannot find name 'anchors'.
ERROR: 6:1    expect  TypeScript@next compile error:
Cannot find name 'anchors'.
ERROR: 7:1    expect  TypeScript@next compile error:
Cannot find name 'anchors'.
ERROR: 9:1    expect  TypeScript@next compile error:
Cannot find name 'anchors'.
ERROR: 11:27  expect  TypeScript@next compile error:
Cannot find name 'AnchorJS'.
ERROR: 13:27  expect  TypeScript@next compile error:
Cannot find name 'AnchorJS'.
```

Comment (plantain-00) [reply types/anchor-js/tslint.json]: @andy-ms

Comment (xt0rted) [reply types/anchor-js/tslint.json]: @andy-ms if I switch the file around to use that then I need to add `import { anchors, AnchorJS } from "anchor-js";` to the top of my test file. Is this correct? I can push my changes for review if it'd help.

Comment (andy-ms) [reply types/anchor-js/tslint.json]: I think I read the file wrong -- `AnchorJS` is the one equivalent to the module. So you should write `export as namespace AnchorJS;`, and also `declare global { const anchors: anchorjs.Anchor; }` (how is that variable used from a module though?). Once the globals are declared you shouldn't need imports in the test file.

**Comment by andy-ms** [comment]
Thanks!

**Pull request merged**